### PR TITLE
feat/eciML1515

### PR DESCRIPTION
**Pull request by BenjaSanchez**
add eciML1515 to master branch

### CODE REVIEW

**Comment by IVANDOMENZAIN** [comment]
Hej @BenjaSanchez, The non-yeast GEMs in ecModels are not compatible with the latest integration of the `model.parameters` functionality in GECKO. To achieve that, I will need to reduce the amount of provided **specific scripts** and also modify the `update GEM script` for each organism.

I think that it would be better if we provide a non-conflicting version of each ecModel update pipeline in the master branch of this repo, what do you think?

**Comment by BenjaSanchez** [comment]
@IVANDOMENZAIN as the `model.parameters` functionality is still not in the master branch of GECKO, this will not create yet an issue. I opened this PR because there is interest here at CfB to soon have the _E. coli_ model running in DeCaF for visualization, so ideally we should have the `.xml` available in master.

Nonetheless I agree with you that for all other models the most sensitive thing is to wait for the new release of GECKO and then adapt each branch before merging to master. In the case of _E. coli_ we can do that on a fix branch. Let me know if you think this is fine.

**Comment by IVANDOMENZAIN** [comment]
I think this is OK then @BenjaSanchez, let's proceed then. Also, people at Minho University have started to use this model for the validation of one of their developed pipelines, however `geckopy` includes hard-coded information for fields such as: `sigma`, `GAM`, `polimerization costs`, etc. Do you think that it could be straightforward to make some of them optional (mostly polimerization costs) and allow the rest of `GECKOpy` to run?

**Comment by BenjaSanchez** [comment]
@IVANDOMENZAIN for the moment we will not use geckopy for `eciML1515`, but include some small routines directly onto DeCaF that skip all of those optional parameters and only focus on the overlay of proteomics data. We will use the iterative method that we have been using so far in GECKO, but translated to python so to be compatible with the DeCaF backend. Will keep you on the loop of further development ;)

In the future I would like to adapt geckopy so that it is generic to any model, with all of those parameters set as elective, and perhaps migrate it to a separate repo (which would make it much more manageable). However I don't have a clear timeline on this yet.

**Comment by IVANDOMENZAIN** [comment]
@BenjaSanchez have you already tested the eciML1515 pipeline locally? If so, are you getting a functional `ecModel`?

**Comment by BenjaSanchez** [comment]
@IVANDOMENZAIN good point, the pipeline does not work locally. Perhaps is best to leave this PR open for now and merge once the new version of GECKO is released. For now we will use the `.xml` file available in this branch :)

**Comment by IVANDOMENZAIN** [comment]
@BenjaSanchez could you let me know about the errors that you're getting? an error message screenshot would be good :)

**Comment by BenjaSanchez** [comment]
@IVANDOMENZAIN two sections of the main script `eciML1515update.m` are commented:
```matlab
% git('clone https://github.com/SysBioChalmers/GECKO.git')
% %Load iML1515 model:
% cd model
% model    = importModel('iML1515.xml');
% cd ..
```
and:
```matlab
%[ecModel,ecModel_batch,version] = enhanceGEM(model,'COBRA');
```
After uncommenting them locally I get towards the end (after the model has been created and saved) the following error:
```matlab
Undefined function or variable 'name'.
Error in eciML1515update (line 22)
moveModelFiles(name)
```
This `name` variable seems to be missing in the script.

**Comment by IVANDOMENZAIN** [comment]
Thanks @BenjaSanchez! I'll fix it.

**Comment by BenjaSanchez** [comment]
@IVANDOMENZAIN great :) something else that I noticed doing some testing in the `eciML1515` model (i.e. no actual constraints) is that it is not able to grow, due to different upper bounds in its exchange reactions. See [this](https://nbviewer.jupyter.org/github/BenjaSanchez/notebooks/blob/master/e_coli_simulations.ipynb) notebook for the analysis (I also checked that the issue was present with Matlab/COBRA toolbox). Any `ecGEM.xml` file should in fact predict the same as its `GEM.xml` counterpart (we could even make it a test by Jenkins), as in that file we are only including additional reactions/metabolites but no additional constraints. As an example, currently `ecYeastGEM.xml` predicts the same growth rate as `yeastGEM.xml` without having to modify in any way the model. As shown in the notebook, I managed to make `eciML1515` grow by:

* Replacing all `Inf` upper bounds by `1000` (to avoid an unbounded problem).
* Blocking at zero all uptake reactions (i.e. exchange reactions with the `_REV` suffix) except the 16 needed to grow by `iML1515`, which I left open between 0 and 1000 (except for glucose which I left between 0 and 10 just like in the original model).

However I think it would be preferable if in the ecModel generation steps we would preserve all bounds from the original model to avoid the back-and-forth. Let me know your thoughts!

File: eciML1515/scripts/fitGAM.m [hunk 1]
```diff
@@ -0,0 +1,93 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% GAM = fitGAM(model)
+% Returns a fitted GAM for the yeast model.
+% 
+% Ivan Domenzain. Last update: 2019-04-24
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+function GAM = fitGAM(model)
```
Comment (BenjaSanchez) [inline eciML1515/scripts/fitGAM.m]: why the need of this function? It is not used in this branch's version of `scaleBioMass.m` so could be skipped entirely

Comment (IVANDOMENZAIN) [reply eciML1515/scripts/fitGAM.m]: Thanks for noticing this, the function has now been removed by 
b7e0131

**Comment by IVANDOMENZAIN** [comment]
> @IVANDOMENZAIN two sections of the main script `eciML1515update.m` are commented:
> 
> ```matlab
> % git('clone https://github.com/SysBioChalmers/GECKO.git')
> % %Load iML1515 model:
> % cd model
> % model    = importModel('iML1515.xml');
> % cd ..
> ```
> 
> and:
> 
> ```matlab
> %[ecModel,ecModel_batch,version] = enhanceGEM(model,'COBRA');
> ```
> 
> After uncommenting them locally I get towards the end (after the model has been created and saved) the following error:
> 
> ```matlab
> Undefined function or variable 'name'.
> Error in eciML1515update (line 22)
> moveModelFiles(name)
> ```
> 
> This `name` variable seems to be missing in the script.

These issues have been addressed by commit 
56d702a, the pipeline works locally and returns the same model that has already been stored in this branch. Thanks @BenjaSanchez! :D

**Comment by IVANDOMENZAIN** [comment]
> this

Such a nice example in the provided notebook! I agree @BenjaSanchez, functional `ecModel` structures should be provided and for me it also sounds reasonable to preserve the original model's bounds in the enzyme-constrained version. In order to implement this, we can create an issue in GECKO referring to this conversation and then we take a thorough look to all functions in GECKO that are modifying the ecModel bounds. If we implement this, then it will be clearer that the `ecModel` structure contains the same original  bounds but just with unconstrained enzymes linked to reactions and, `ecModel_batch` incorporates a global constraint in enzymes usage but also the bounds that are introduced by `changeMedia_batch.m`

**Comment by mihai-sysbio** [comment]
@IVANDOMENZAIN resolved the conflicts manually by choosing a mix of this branch and `main`.

**Pull request merged**